Introduce allNullOr methods

VincentLanglet opened the pull request:
Hi @Nyholm 

I recently had the need of a `allNullOrScalar` method to check (and help my static analysis tool to understand) a variable is `array<bool|int|float|string|null>`.

To keep consistency I added all the `allNullOr` possible methods.

zerkms commented:
Did you manually patch the `src/Mixin.php`? If that's the case - have a look at https://github.com/webmozarts/assert/blob/master/bin/src/MixinGenerator.php

The mixin is fully code-generated and is not intended to be changed manually.

I think before we did have a note about it in the header, or may be I'm confusing it with something else.

VincentLanglet commented:
> Did you manually patch the `src/Mixin.php`? If that's the case - have a look at https://github.com/webmozarts/assert/blob/master/bin/src/MixinGenerator.php

Yes I did manually patch the Mixing file :'(

> The mixin is fully code-generated and is not intended to be changed manually.

I updated the PR.

> I think before we did have a note about it in the header, or may be I'm confusing it with something else.

The header say
>  * This trait provides nurllOr*, all* and allNullOr* variants of assertion base methods.
> * Do not use this trait directly: it will change, and is not designed for reuse.

It said "Do not use directly", not "do not change directly".
Might be worth to explain in the header how to update this file.

VincentLanglet commented:
Friendly ping @BackEndTea @webmozart, since I'm not sure who's currently maintaining this :)

VincentLanglet commented:
@Nyholm @BackEndTea @webmozart,
Hi, who is currently maintaining the library currently ? Do you have some time to take a look at the PR ? :)

Big thanks

VincentLanglet commented:
Do you have some time to take a look at this PR @BackEndTea ? :)

BackEndTea commented:
I fixed a small bug with the `allNullOr*` generation and added test for the new methods.

I'll merge this is everything looks good

VincentLanglet commented:
> I fixed a small bug with the `allNullOr*` generation and added test for the new methods.
> 
> I'll merge this is everything looks good

Thanks a lot

VincentLanglet commented:
> I fixed a small bug with the allNullOr* generation and added test for the new methods.
> I'll merge this is everything looks good

@BackEndTea I might have misunderstand. Is everything good for a merge/release or do I need to do some fixes ?

BackEndTea commented:
Thank you for the contribution @VincentLanglet

herndlm commented:
I can add support for this in the phpstan extension after release. Except you want to @VincentLanglet :)

BackEndTea commented:
@herndlm The release has been tagged
